app/models/oclc/record.rb

```diff
@@ -0,0 +1,250 @@
+# frozen_string_literal: false
+
+module Oclc
+  # rubocop:disable Metrics/ClassLength
+  class Record
+    attr_reader :record
+    def initialize(marc_record:)
+      @record = marc_record
+    end
+
+    def generally_relevant?
+      !juvenile? && !audiobook? && !published_in_us_uk_or_canada? && monograph? && within_last_two_years?
+    end
+
+    def relevant_to_selector?(selector:)
+      call_number_in_range_for_selector?(selector:) || subject_relevant_to_selector?(selector:)
+    end
+
+    def call_number_in_range_for_selector?(selector:)
+      return false unless class_relevant_to_selector?(selector:)
+
+      selector.call_number_ranges.map do |range|
+        next false unless lc_class == range[:class]
+
+        next true if lc_number.between?(range[:low_num], range[:high_num])
+
+        false
+      end.include?(true)
+    end
+
+    def class_relevant_to_selector?(selector:)
+      return true if selector.classes.include?(lc_class)
+
+      false
+    end
+
+    def subject_relevant_to_selector?(selector:)
+      subjects.any? { |subject| subject.downcase.match?(/#{selector.subjects.join('|')}/) }
+    end
+
+    def author
+      # The first available author field
+      auth_field = record.fields(%w[100 110 111]).try(:[], 0)
+      return '' if auth_field.blank?
+
+      auth_tag = auth_field.tag
+      subf_to_skip = auth_subfields_to_skip(auth_tag)
+      targets = auth_field.subfields.reject do |subfield|
+        subf_to_skip.include?(subfield.code)
+      end
+
+      targets.map { |field| scrub_string(field.value) }.join(' ')
+    end
+
+    def description
+      return '' unless record['300']
+
+      record['300'].subfields.map(&:value).join(' ')
```

Comment:
Would record['300'].to_s accomplish a similar purpose?

Reply:
For one of test objects, the output of the above code is `"volumes <1, 2-3, 5-14>; 25 cm."`

The output of `record['300'].to_s` is `"300    $a volumes <1, 2-3, 5-14>; $c 25 cm."`, which I think does not serve our purpose of human readability in the context of a CSV.